### PR TITLE
fix(typing): fix DiscordUser and Presence Update typing 

### PR DESCRIPTION
**Pull request by H01001000**
As shown in [discord docs](https://discord.com/developers/docs/topics/gateway#presence-update), the presence update's user can be partial. This PR fix the typing and the transformer

### CODE REVIEW

**Review by lts20050703 — CHANGES_REQUESTED** [review]
This PR has unrelated changes. Please revert unrelated changes

**Comment by H01001000** [comment]
I remove the commit and changed to use PickPartial instead

File: transformers/member.ts [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    premiumType: payload.premium_type,
-    publicFlags: payload.public_flags,
+    username: "username" in payload ? payload.username : undefined,
+    discriminator: "discriminator" in payload ? payload.discriminator : undefined,
+    avatar: "avatar" in payload ? bot.utils.iconHashToBigInt(payload.avatar) : undefined,
+    locale: "locale" in payload ? payload.locale : undefined,
+    email: "email" in payload ? payload.email : undefined,
+    flags: "flags" in payload ? payload.flags : undefined,
+    premiumType: "premium_type" in payload ? payload.premium_type : undefined,
+    publicFlags: "public_flags" in payload ? payload.public_flags : undefined,
```
Comment (lts20050703) [inline transformers/member.ts:17]:
```suggestion
    username: payload.username ?? undefined,
    discriminator: payload.discriminator ?? undefined,
    avatar: payload.avatar ? bot.utils.iconHashToBigInt(payload.avatar) : undefined,
    locale: payload.locale ?? undefined,
    email: payload.email ?? undefined,
    flags: payload.flags ?? undefined,
    premiumType: payload.premium_type ?? undefined,
    publicFlags: payload.public_flags ?? undefined,
```

**Review by lts20050703 — APPROVED** [review]

**Review by lts20050703 — CHANGES_REQUESTED** [review]
Please run `deno fmt`

**Review by lts20050703 — APPROVED** [review]

**Pull request closed without merging**